internal/processor/attraction/attraction.go

```diff
@@ -0,0 +1,284 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package attraction
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/internal/processor/filterhelper"
+)
+
+// Settings
+type Settings struct {
+	// Actions specifies the list of attributes to act on.
+	// The set of actions are {INSERT, UPDATE, UPSERT, DELETE}.
+	// This is a required field.
+	Actions []ActionKeyValue `mapstructure:"actions"`
+}
+
+// ActionKeyValue specifies the attribute key to act upon.
+type ActionKeyValue struct {
+	// Key specifies the attribute to act upon.
+	// This is a required field.
+	Key string `mapstructure:"key"`
+
+	// Value specifies the value to populate for the key.
+	// The type of the value is inferred from the configuration.
+	Value interface{} `mapstructure:"value"`
+
+	// A regex pattern  must be specified for the action EXTRACT.
+	// It uses the attribute specified by `key' to extract values from
+	// The target keys are inferred based on the names of the matcher groups
+	// provided and the names will be inferred based on the values of the
+	// matcher group.
+	// Note: All subexpressions must have a name.
+	// Note: The value type of the source key must be a string. If it isn't,
+	// no extraction will occur.
+	RegexPattern string `mapstructure:"pattern"`
+
+	// FromAttribute specifies the attribute from the span to use to populate
+	// the value. If the attribute doesn't exist, no action is performed.
+	FromAttribute string `mapstructure:"from_attribute"`
+
+	// Action specifies the type of action to perform.
+	// The set of values are {INSERT, UPDATE, UPSERT, DELETE, HASH}.
+	// Both lower case and upper case are supported.
+	// INSERT -  Inserts the key/value to spans when the key does not exist.
+	//           No action is applied to spans where the key already exists.
```

Comment:
should we replace all the `spans` wording to `attributes`?

Reply:
Done.